### ISSUE TITLE
[BUG/DEVELOP] When Issuing a Close All Command, Not All Tabs Are Closed

### ISSUE THREAD

**Issue by nathanlesage** [Zettlr/Zettlr#1265]
## Description

@anoopengineer stumbled upon this during implementation of PR #1250:

> I added that specifically to handle an edge condition. Steps to reproduce:
>
> 1. Open 3-4 tabs
> 2. Close Zettlr
> 3. Relaunch Zettlr
>
> Zettlr will come up with previous tabs with one tab highlighted.  
> Right click on the highlighted tab and select close all  
> You'll see that the last tab still remains open.
>
> I added setting `_currentHash` to `null` to handle this. But your comment made me dig in a bit deeper. To me it looks like there is a bug elsewhere in the system. When closeAll fires off IPC for n files, only n-1 calls to `close()` are received causing last file to remain open. This is happening only when you close Zettlr with many tabs and relaunch it. I am not sure what might be causing this.

## Reproducing

See the comment above

## Expected behaviour

Send closing commands to all tabs

## Platform
 - OS and version: macOS 10.15.6 and Arch Linux
 - Zettlr Version: 1.8.x development branch

**Comment by Kangie** [comment]
Interestingly, this is inconsistent on Win10 (20.04). 

Sometimes I experience this bug when switching to develop from 1.7.5 (new features vs ability to export as PDF using my template).

If I have a lot of tabs open, I usually still have one open after using the 'close all' command, but sometimes all the tabs close. No idea why!

**Comment by nathanlesage** [comment]
This is what we have to investigate :D

**Comment by nathanlesage** [comment]
What I found: indeed the right amount of close-calls are being made, and all four are being handled in main. Then, the handling command successfully issues close-commands to the FSAL. This brought me to assume that there is a race condition, as all IPC calls are asynchronous, and I implemented the close all functionality in main directly, instead of sending a bunch of disparate events.

However, this still not yielded success. After inspecting the logic in the editor, however, I found several wrong conditions.

In the end, I simplified the code of `syncFiles` and hard-removed files that were no longer open, thus additionally increasing the speed of sync actions. I added further simplifications to clean up the mess I made.
